py/datastore_service.py

```diff
@@ -103,11 +104,44 @@ def handle_request(self, post):
     #get the reporter data
     reports = self.parse_reports(post)
 
-    #TODO: build a query
-    #for report in reports:
+    try:
+      # check and see if prepared statement exists...if not, create it
```

Comment:
this happens on every request! if we can, i would move this to make_locals above so that it would only be done once.